backend-rust/src/indexer.rs

```diff
@@ -2263,13 +2263,14 @@ struct PreparedModuleDeployed {
 impl PreparedModuleDeployed {
     async fn prepare(
         node_client: &mut v2::Client,
-        data: &BlockData,
         module_reference: sdk_types::hashes::ModuleReference,
     ) -> anyhow::Result<Self> {
-        let block_height = data.finalized_block_info.height;
-
+        // The `get_module_source` query on old blocks are currently not performing
+        // well in the node. We query on the `lastFinal` block here as a result (https://github.com/Concordium/concordium-scan/issues/534).
+        // The connected node should be caught-up to the top of the chain to make use of
+        // the performance gain.
```

Comment:
I think whether the node is caught up is not really relevant. As long as the module was in some finalized block, then looking up in the last finalized block is going to be the fastest way to get it. (I don't think we are concerned with non-finalized blocks here at all.)